app/services/bill-runs/two-part-tariff/submit-remove-bill-run-licence.service.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+/**
+ * Orchestrates removing a licence from a bill run whilst it is at the review stage
+ * @module SubmitRemoveBillRunLicenceService
+ */
+
+const { db } = require('../../../../db/db.js')
+
+/**
+ * Orchestrates removing a licence from a bill run whilst it is at the review stage. It does this by deleting all of the
+ * persisted data relating to the licence from the review tables. The licence will then be flagged for 2PT supplementary
+ * billing. If after removing a licence the bill run is empty, the bill run status will be set to set to `empty` and
+ * `true` returned so that the user is redirected back to the Bill runs page rather than Review licences.
+ *
+ * @param {string} billRunId - The UUID of the bill run that the licence is in
+ * @param {string} licenceId UUID of the licence to remove from the bill run
+ * @param {Object} sessionManager - The Hapi `request.yar` session manager passed on by the controller
+ */
+async function go (billRunId, licenceId, yar) {
+  await _removeChargeElementReturns(billRunId, licenceId)
+  await _removeReturns(billRunId, licenceId)
+  await _removeChargeElements(billRunId, licenceId)
+  await _removeChargeReferences(billRunId, licenceId)
+  await _removeChargeVersions(billRunId, licenceId)
+  await _removeLicence(billRunId, licenceId)
+
+  const licenceRef = await _setSupBillingTrueAndReturnRef(licenceId)
+
+  const allLicencesRemoved = await _allLicencesRemoved(billRunId, licenceRef, yar)
+
+  return allLicencesRemoved
+}
+
+async function _allLicencesRemoved (billRunId, licenceRef, yar) {
+  const { licenceCount } = await db
+    .count('id AS licenceCount')
+    .from('reviewLicences')
+    .where('billRunId', billRunId)
+    .first()
+
+  if (licenceCount === 0) {
+    await db
+      .update('status', 'empty')
+      .from('billRuns')
+      .where('id', billRunId)
+
+    return true
+  }
+
+  // NOTE: The banner message is only set if licences remain in the bill run. This is because if there are no longer any
+  // licences remaining in the bill run the user is redirected to the "Bill runs" page instead of "Review licences". As
+  // the banner isn't displayed on the "Bill runs" page the message would remain in the cookie which could cause issues.
+  yar.flash('banner', `Licence ${licenceRef} removed from the bill run.`)
```

Comment:
We shouldn't be doing this in this method as it is a clear violation of 'Single responsibility principle' from [SOLID](https://en.wikipedia.org/wiki/SOLID).

We can use the result this function returns in the `go()` to determine if the flash banner needs setting. That also means we do not need to pass yar or the licence ref into this function.